Rubberduck.UnitTesting/UnitTesting/TestCodeGenerator.cs

```diff
@@ -0,0 +1,231 @@
+            }
+
+            return string.Format(TestModuleTemplate,
+                _isAccess ? AccessCompareOption : string.Empty,
+                declaration,
+                initialization);
+        }
+
+        public string GetNewTestMethodCode(IVBComponent component)
+        {
+            return string.Join(Environment.NewLine,
+                _indenter.Indent(string.Format(TestMethodTemplate, GetNextTestMethodName(component))));
+        }
+
+        public string GetNewTestMethodCodeErrorExpected(IVBComponent component)
+        {
+            return string.Join(Environment.NewLine,
+                _indenter.Indent(string.Format(TestMethodExpectedErrorTemplate, GetNextTestMethodName(component))));
+        }
+
+        private string GetNewTestStubMethod(Declaration procedure)
+        {
+            var name = string.Empty;
+
+            switch (procedure.DeclarationType)
+            {
+                case DeclarationType.Procedure:
+                case DeclarationType.Function:
+                    name = $"{procedure.IdentifierName}{TestMethodBaseName}";
+                    break;
+                case DeclarationType.PropertyGet:
+                    name = $"Get{procedure.IdentifierName}{TestMethodBaseName}";
+                    break;
+                case DeclarationType.PropertyLet:
+                    name = $"Let{procedure.IdentifierName}{TestMethodBaseName}";
+                    break;
+                case DeclarationType.PropertySet:
+                    name = $"Set{procedure.IdentifierName}{TestMethodBaseName}";
+                    break;
+            }
+
+            return string.Format(TestMethodTemplate, name);
+        }
+
+        private string GetNextTestModuleName(IVBProject project)
+        {
+            var names = new HashSet<string>(project.ComponentNames().Where(module => module.StartsWith(TestModuleBaseName)));
+
+            var index = 1;
+            while (names.Contains($"{TestModuleBaseName}{index}"))
+            {
+                index++;
+            }
+
+            return $"{TestModuleBaseName}{index}";
+        }
+
+        private string GetNextTestMethodName(IVBComponent component)
+        {
+            var names = new HashSet<string>(_state.DeclarationFinder.Members(component.QualifiedModuleName)
+                .Select(test => test.IdentifierName).Where(decl => decl.StartsWith(TestMethodBaseName)));
+
+            var index = 1;
+            while (names.Contains($"{TestMethodBaseName}{index}"))
+            {
+                index++;
+            }
+
+            return $"{TestMethodBaseName}{index}";
+        }
+
+        private IEnumerable<Declaration> GetDeclarationsToStub(Declaration parentDeclaration)
+        {
+            return _state.DeclarationFinder.Members(parentDeclaration)
+                .Where(d =>
+                            Equals(d.ParentDeclaration, parentDeclaration) && d.Accessibility == Accessibility.Public &&
```

Comment:
Agreed. That change will need to happen when #4658 is addressed.

Reply:
I'm an idiot - this _doesn't_ need to be addressed with #4658 - this is looking for methods to generate test stubs for, _not_ tests.  We're creating the tests in a different module, so they have to be public - otherwise the test stub won't be able to call it.  RD don't have to respect the accessibility, but VBA does...